apps/frontend/src/libs/hooks/use-popover-position/use-popover-position.hook.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from "~/libs/hooks/hooks.js";
+
+type Properties = {
+	left: number;
+	top: number;
+};
+
+const usePopoverPosition = <T extends HTMLElement>(
+	reference: React.RefObject<T>,
+	isActive: boolean,
+): Properties => {
+	const [position, setPosition] = useState<Properties>({ left: 0, top: 0 });
+
+	useEffect(() => {
+		if (reference.current && isActive) {
+			const targetRect = reference.current.getBoundingClientRect();
+			const { left, top } = targetRect;
+
+			setPosition({ left, top });
+		}
+	}, [reference, isActive]);
```

Comment:
I also added `const scrollOffset = window.scrollY || document.documentElement.scrollTop;` so the popover appears correctly positioned when the user has scrolled down the page.